staging/src/k8s.io/apiserver/pkg/endpoints/handlers/create.go

```diff
@@ -129,6 +130,15 @@ func createHandler(r rest.NamedCreater, scope *RequestScope, admit admission.Int
 		audit.LogRequestObject(ae, obj, scope.Resource, scope.Subresource, scope.Serializer)
 
 		userInfo, _ := request.UserFrom(ctx)
+		// On create, get name from new object if unset
+		if len(name) == 0 {
+			_, name, err = scope.Namer.ObjectName(obj)
+			if err == nil {
+				if metadata, err := meta.Accessor(obj); err == nil {
```

Comment:
why is this modifying the object? I only expected to extract the name from the object